### PR TITLE
deprecate outdated param for deploying k8s on ubuntu 

### PR DESCRIPTION
**Pull request by dalanlan**
@resouer  @WIZARD-CXY

### CODE REVIEW

**Comment by WIZARD-CXY** [comment]
I think auto provision namespace is handy.

**Comment by dalanlan** [comment]
It would soon be deprecated though. See also [here](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/admin/admission-controllers.md#namespaceautoprovision-deprecated).

File: docs/user-guide/service-accounts.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ kind: ServiceAccount
 metadata:
   name: build-robot
 EOF
-$ kubectl create -f /tmp/serviceaccount.json
+$ kubectl create -f /tmp/serviceaccount.yaml
```
Comment (WIZARD-CXY) [inline docs/user-guide/service-accounts.md]: I'm not sure this change is ok ?

Comment (dalanlan) [inline docs/user-guide/service-accounts.md]: It's a typo, i'll give you that.

**Comment by WIZARD-CXY** [comment]
LGTM

**Comment by mikedanese** [comment]
LGTM

**Pull request merged**